typed-racket-lib/typed-racket/typecheck/tc-literal.rkt

```diff
@@ -124,8 +125,21 @@
                  [ks (hash-map h (lambda (x y) (tc-literal x)))]
                  [vs (hash-map h (lambda (x y) (tc-literal y)))])
             (make-Hashtable (generalize (apply Un ks)) (generalize (apply Un vs))))])]
+    [(~var i (3d prefab-struct-key))
+     (tc-prefab (syntax-e #'i) expected)]
     [_ Univ]))
 
-
-
-
+;; Typecheck a prefab struct literal
+(define (tc-prefab struct-inst expected)
+  (define expected-ts
+    (match (and expected (resolve expected))
+      [(Prefab: _ ts) (in-sequence-forever (in-list ts) #f)]
+      [_ (in-cycle (in-value #f))]))
```

Comment:
This just throws away the expected type when it's wrong. Can't that miss errors?